percona/controller/pgcluster/controller.go

```diff
@@ -50,42 +53,44 @@ func (r *PGClusterReconciler) SetupWithManager(mgr manager.Manager) error {
 func (r *PGClusterReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logging.FromContext(ctx)
 
-	perconaPGCluster := &v2beta1.PerconaPGCluster{}
-	if err := r.Client.Get(ctx, request.NamespacedName, perconaPGCluster); err != nil {
+	rr := ctrl.Result{RequeueAfter: 5 * time.Second}
```

Comment:
@egegunes was there a reason why we didn't return the result with `RequeueAfter`, or just something that was overlooked?

Reply:
It was on purpose. Using `RequeueAfter` by default is not a good design, since it causes unneeded reconciliations. Our controller should reconcile objects only if something updated the objects it watches. Unfortunately we used it from the start in other operators and depended on it but I don't think we should repeat the same mistake here. Why do you need to have it?

Reply:
For example, when you deploy a cluster without exposing PGbouncer and then expose it, without this `status.host` will never get the ingress IP since we don't reconcile.

Reply:
the solution is not using `RequeueAfter` but watching the `Service` object in our controller

Reply:
Yes, now I see that.